EVG-16693 Remove project param from delete CQ route

hadjri opened the pull request:
[EVG-16693](https://jira.mongodb.org/browse/EVG-16693)

### Description 
Modification to the commit queue delete route, making it accept just a single commit queue item and do a project DB lookup on the server side rather than on the client side where it is currently invalid.
### Testing 
Testing in staging env (attached)

Kimchelly reviewed (approved):
Since we're changing the path for a route in the public API, does the wiki need to be updated? I didn't see any of the commit queue routes in it, but just wanted to double check.

hadjri commented:
> Since we're changing the path for a route in the public API, does the wiki need to be updated? I didn't see any of the commit queue routes in it, but just wanted to double check.

The commit queue CLI has a separate wiki page which I've updated (i.e. removed the project parameter from)

https://github.com/evergreen-ci/evergreen/wiki/Commit-Queue#delete

hadjri commented on rest/route/commit_queue.go:
```diff
@@ -71,9 +71,15 @@ func (cq commitQueueDeleteItemHandler) Factory() gimlet.RouteHandler {
+	if requestedPatch == nil {
+		return errors.New("commit queue item not found")
+	}
+	cq.project = requestedPatch.Project
```
The logic in Run that handles sending a status to GitHub relies on ProjectRef fields so I think we actually need to keep the project ref retrieval: https://github.com/evergreen-ci/evergreen/blob/main/rest/route/commit_queue.go#L107

ablack12 replied:
Ah good catch -- I do think we can just wait to get the project ref until that if statement then, since it's not relevant until then.

hadjri commented on rest/route/service.go:
```diff
@@ -82,7 +82,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/builds/{build_id}/annotations").Version(2).Get().Wrap(requireUser, viewAnnotations).RouteHandler(makeFetchAnnotationsByBuild())
 	app.AddRoute("/commit_queue/{project_id}").Version(2).Get().Wrap(requireUser, viewTasks).RouteHandler(makeGetCommitQueueItems())
-	app.AddRoute("/commit_queue/{project_id}/{item}").Version(2).Delete().Wrap(requireUser, addProject, requireCommitQueueItemOwner, editTasks).RouteHandler(makeDeleteCommitQueueItems(env))
+	app.AddRoute("/commit_queue/{patch_id}").Version(2).Delete().Wrap(requireUser, addProject, requireCommitQueueItemOwner, editTasks).RouteHandler(makeDeleteCommitQueueItems(env))
```
Checking in Splunk there aren't a super high amount of requests to this endpoint - for those that do exist I think it's safe to assume they come from the CLI and not manual requests
